### PR TITLE
PRMDR-879: Implement lambda layers and cleanup script into workflow

### PR DESCRIPTION
**Pull request by abbas-khan10**

### CODE REVIEW

File: .github/workflows/terraform-destroy-dev-environments-cron.yml [hunk 1]
```diff
@@ -10,13 +10,29 @@ permissions:
   contents: read  # This is required for actions/checkout
     
 jobs:
+  cleanup_versions_process:
+    name: Cleanup Versions Process
+    uses: ./.github/workflows/cleanup-appconfig-and-lambda-layer-versions.yml
+    strategy:
+      matrix:
+        # Update with new sandbox environments here
+        sandbox-name: [ ndra, ndrb, ndrc ]
```
Comment (NogaNHS) [inline .github/workflows/terraform-destroy-dev-environments-cron.yml:19]: Why are we excluding ndrd?

Comment (abbas-khan10) [reply .github/workflows/terraform-destroy-dev-environments-cron.yml:19]: I didn't get chance to discuss this with Scott last Friday, it had been excluded from the matrix on line 36 also also which makes me think ndrd was been kept aside for demo purposes. So for now just keeping consistency with the other matrixes and when confirmed we can add ndrd back in

File: scripts/cleanup_versions.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        for version in excess_hosted_config_versions:
+            response = self.appconfig_client.delete_hosted_configuration_version(
+                ApplicationId=version['ApplicationId'],
+                ConfigurationProfileId=version['ConfigurationProfileId'],
+                VersionNumber=version['VersionNumber']
+            )
+            if response["ResponseMetadata"]["HTTPStatusCode"] == 204:
+                successful_deletes.append(response)
+
+        if total_untracked_versions:
```
Comment (NogaNHS) [inline scripts/cleanup_versions.py:60]: Do we need this if? we are checking in line 46 `if not total_untracked_versions`

File: scripts/cleanup_versions.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+        current_hosted_configuration_versions = self.appconfig_client.list_hosted_configuration_versions(
+            ApplicationId=application_id,
+            ConfigurationProfileId=config_profile_id
+        )
+
+        return current_hosted_configuration_versions["Items"]
+
+    def delete_hosted_configuration_versions(self):
+        excess_hosted_config_versions = self.get_hosted_configuration_versions()
```
Comment (NogaNHS) [inline scripts/cleanup_versions.py:42]: I'm probably missing something, but how does it exclude the original version?

Comment (abbas-khan10) [reply scripts/cleanup_versions.py:42]: No you haven't missed anything good spot, but it was done on purpose. It doesn't look like the TF state cares about tracking the hosted configuration versions as they're not something that are really a usable feature in terraform (which is why we're in this predicament in the first place), and so removing every hosted configuration version doesn't cause any issues when it comes to the terraform destroy. We'd just be adding unnecessary filtering logic to remove the original version

File: scripts/cleanup_versions.py (hunk 2 above)
Comment (abbas-khan10) [reply scripts/cleanup_versions.py:60]: No this can be removed, cheers

**Review by NogaNHS — APPROVED** [review]

**Pull request merged**